### PR TITLE
Improves mulebot movement code

### PR DESCRIPTION
**Pull request by Sarchutar**
## About The Pull Request

Ports https://github.com/tgstation/tgstation/pull/51179.
On the Things to Port List: https://github.com/BeeStation/BeeStation-Hornet/projects/2#card-38817620

Original description:

> Unfucks mulebot timer bullshit, makes some math cleaner, and removes an old fuckup in how the code got translated from sleep to the if check we have now.

## Why It's Good For The Game

> Hardens the code against timers dying, cleans shit up, and kickstarts the skogcoin market.

## Changelog
:cl:LemonInTheDark
fix: Mulebots no longer go insane when shit starts to lag 
/:cl:

### CODE REVIEW

**Comment by ike709** [comment]
Did you test this?

**Comment by Sarchutar** [comment]
> Did you test this?

I did test this, and the mulebots still work like they should.

File: code/modules/mob/living/simple_animal/bot/mulebot.dm [hunk 1]
```diff
@@ -458,25 +460,17 @@
...
+		if(!speed)//Devide by zero man bad
+			return
+		num_steps = round(10/speed) //10, 5, or 3 steps, depending on how many wires we have cut
+		if(mode != BOT_IDLE)
+			START_PROCESSING(SSfastprocess, src)
+
+/mob/living/simple_animal/bot/mulebot/process()
+	if(num_steps <= 0)
+		STOP_PROCESSING(SSfastprocess, src)
+		return
```
Comment (park66665) [inline code/modules/mob/living/simple_animal/bot/mulebot.dm]:
```suggestion
		return PROCESS_KILL
```

**Review by park66665 — APPROVED** [review]
MULEbots work as intended. All clear. This PR is good to go.

**Pull request merged**